File: UI/travle_core/lib/src/models/destination_insert_request.dart
import 'package:json_annotation/json_annotation.dart';

import 'destination_image_request.dart';

part 'destination_insert_request.g.dart';

/// A curator/organizer's destination submission (mirrors the backend
/// `DestinationInsertRequest`). The submitter comes from the JWT and the status
/// always starts Pending — neither is sent here. Coordinates are entered directly
/// for now (a map picker replaces the manual fields later).
@JsonSerializable()
class DestinationInsertRequest {
  DestinationInsertRequest({
    required this.name,
    required this.description,
    required this.categoryId,
    required this.cityId,
    required this.latitude,
    required this.longitude,
    this.tagIds = const [],
    this.images = const [],
  });

  final String name;
  final String description;
  final int categoryId;
  final int cityId;
  final double latitude;
  final double longitude;
  final List<int> tagIds;
  final List<DestinationImageRequest> images;

  factory DestinationInsertRequest.fromJson(Map<String, dynamic> json) =>
      _$DestinationInsertRequestFromJson(json);

  Map<String, dynamic> toJson() => _$DestinationInsertRequestToJson(this);
}

File: Backend/Travle.Model/Requests/DestinationInsertRequest.cs
namespace Travle.Model.Requests
{
    /// <summary>
    /// A curator/organizer's destination submission. The submitter is taken from the JWT (never the
    /// client) and the destination always starts <c>Pending</c> — neither is accepted here. Category,
    /// city and tags are FK ids resolved server-side; coordinates are entered directly for now (a map
    /// picker replaces the manual fields later — see 07 §8).
    /// </summary>
    public class DestinationInsertRequest
    {
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;

        public int CategoryId { get; set; }
        public int CityId { get; set; }

        public double Latitude { get; set; }
        public double Longitude { get; set; }

        public List<int> TagIds { get; set; } = new List<int>();
        public List<DestinationImageRequest> Images { get; set; } = new List<DestinationImageRequest>();
    }
}
